pkg/controllers/management/auth/project_cluster/common.go

```diff
@@ -158,3 +212,56 @@ func newOwnerReference(obj runtime.Object) (metav1.OwnerReference, error) {
 		UID:        metadata.GetUID(),
 	}, nil
 }
+
+// needsPSARole ensure that given the project name, it needs a psa role to work properly.
+func needsPSARole(projectName, projectNamespace string, prtbLister v32.ProjectRoleTemplateBindingCache, rtLister v32.RoleTemplateCache) (bool, error) {
+	// lookup the roletemplate(s) associated with the project name
+	// to check if those contain updatepsa verb
+	roleTemplates, err := roleTemplatesLookup(projectName, projectNamespace, prtbLister, rtLister)
+	if err != nil {
+		return false, err
+	}
+
+	return psautils.IsPSAAllowed(roleTemplates), nil
+}
+
+// roleTemplatesLookup returns the list of roletemplates associated to the project.
+func roleTemplatesLookup(projectName, projectNamespace string, prtbLister v32.ProjectRoleTemplateBindingCache, rtLister v32.RoleTemplateCache) ([]*v3.RoleTemplate, error) {
+	// list all the prtbs in the project namespace
+	prtbs, err := prtbLister.List(projectNamespace, labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	if len(prtbs) == 0 {
+		return nil, fmt.Errorf("PRTBs not found")
```

Comment:
is this really an error? or would it be better to return nil or an empty `RoleTemplate` slice? I'm concern we return an error, which will keep try to reconcile if there are not PRTBs for a Project.

Reply:
yes, I agree with you. This could potentially flood the reconciler.
I'm gonna remove it. Thanks